crates/js_api/src/gui/mod.rs

```diff
@@ -61,38 +48,53 @@ pub struct DotrainOrderGui {
     #[serde(skip)]
     state_update_callback: Option<js_sys::Function>,
 }
+
 #[wasm_bindgen]
 impl DotrainOrderGui {
-    #[wasm_bindgen(js_name = "getDeploymentKeys")]
-    pub async fn get_deployment_keys(dotrain: String) -> Result<DeploymentKeys, GuiError> {
+    #[wasm_bindgen(constructor)]
+    pub fn new() -> DotrainOrderGui {
+        Self {
+            dotrain_order: DotrainOrder::dummy(),
+            selected_deployment: "".to_string(),
+            field_values: BTreeMap::new(),
+            deposits: BTreeMap::new(),
+            state_update_callback: None,
+        }
+    }
+}
+
+#[wasm_export]
+impl DotrainOrderGui {
+    #[wasm_export(js_name = "getDeploymentKeys", unchecked_return_type = "string[]")]
+    pub async fn get_deployment_keys(dotrain: String) -> Result<Vec<String>, GuiError> {
         let dotrain_order = DotrainOrder::new(dotrain, None).await?;
-        let keys = GuiCfg::parse_deployment_keys(dotrain_order.dotrain_yaml().documents.clone())?;
-        Ok(DeploymentKeys(keys))
+        Ok(GuiCfg::parse_deployment_keys(
+            dotrain_order.dotrain_yaml().documents.clone(),
+        )?)
     }
 
-    #[wasm_bindgen(js_name = "chooseDeployment")]
+    #[wasm_export(js_name = "chooseDeployment", unchecked_return_type = "void")]
     pub async fn choose_deployment(
+        &mut self,
         dotrain: String,
         deployment_name: String,
         state_update_callback: Option<js_sys::Function>,
-    ) -> Result<DotrainOrderGui, GuiError> {
+    ) -> Result<(), GuiError> {
         let dotrain_order = DotrainOrder::new(dotrain, None).await?;
 
         let keys = GuiCfg::parse_deployment_keys(dotrain_order.dotrain_yaml().documents.clone())?;
         if !keys.contains(&deployment_name) {
             return Err(GuiError::DeploymentNotFound(deployment_name.clone()));
         }
 
-        Ok(Self {
-            dotrain_order,
-            selected_deployment: deployment_name.clone(),
-            field_values: BTreeMap::new(),
-            deposits: BTreeMap::new(),
-            state_update_callback,
-        })
+        self.dotrain_order = dotrain_order;
+        self.selected_deployment = deployment_name;
+        self.state_update_callback = state_update_callback;
+
+        Ok(())
     }
 
-    #[wasm_bindgen(js_name = "getGuiConfig")]
+    #[wasm_export(js_name = "getGuiConfig", unchecked_return_type = "GuiCfg")]
```

Comment:
apparently `unchecked_return_type` isn't necessary when the Result::Ok is the same type. according to @rouzwelt